### DISCUSSION TITLE
Is it possible to make this extension work with Python 3.7?

### DISCUSSION THREAD

**Discussion by abceleung** [microsoft/vscode-mypy#187] [Q&A] [answered]
Before this extension, there were some VSCode mypy settings that allowed me to use mypy with Python 3.7. 

Now these settings are gone (VSCode docs just tell me to install this extension), and this extension only works with Python >= 3.8. 

What can I do (besides upgrading Python)? Thanks!

**Comment by karthiknadig** [comment, marked as answer]
@abceleung Yes. You have two options:
1. You can use older version of the extension 2023.2.0
2. If you have your project cannot be upgraded, but you have newer python in your machine then you can do this. `mypy-type-checker.interpreter` to python-3.8 or greater, and set `mypy-type-checker.path` to ["python3.7", "-m", "mypy"]. This will allow you to use newer version of the extension with older mypy versions.

**Comment by abceleung** [comment]
> @abceleung Yes. You have two options:
> 
>     1. You can use older version of the extension 2023.2.0
> 
>     2. If you have your project cannot be upgraded, but you have newer python in your machine then you can do this. `mypy-type-checker.interpreter` to python-3.8 or greater, and set `mypy-type-checker.path` to ["python3.7", "-m", "mypy"]. This will allow you to use newer version of the extension with older mypy versions.

Thanks. Could you tell me where can I find the older version of the extension?

Edit: Right-click on the installed extension and select `Install Another Version`

**Reply by karthiknadig** [reply]
You can do it using `Install Another Version`:
![image](https://github.com/microsoft/vscode-mypy/assets/3840081/6f2a1b2e-1a83-4194-9fec-a9ea9a8f4450)

**Reply by abceleung** [reply]
Yep, I discovered that right after I made the comment, lol. 
Thanks!
